modules/schedule_module/php/schedule_module.class.inc

```diff
@@ -0,0 +1,120 @@
+<?php declare(strict_types=1);
+/**
+ * File contains the Schedule Module class.
+ *
+ * PHP Version 7
+ *
+ * @category   Core
+ * @package    Main
+ * @subpackage Core
+ * @author     Shen Wang <wangshen.mcin@gmail.com>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+namespace LORIS\schedule_module;
+
+/**
+ * The Schedule Module provides the menu filter for this module.
+ *
+ * @category   Core
+ * @package    Main
+ * @subpackage Core
+ * @author     Shen Wang <wangshen.mcin@gmail.com>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class Schedule_Module extends \DataFrameworkMenu
+{
+    /**
+     * Tells the base class that this page's provisioner can support
+     * the UserSiteMatch filter.
+     *
+     * @return bool true
+     */
+    public function useSiteFilter(): bool
+    {
+        return false;
+    }
+
+    /**
+     * Tells the base class that this page's provisioner can support
+     * the UserProjectMatch filter.
+     *
+     * @return bool false
+     */
+    public function useProjectFilter() : bool
+    {
+        return false;
+    }
+    /**
+     * Tells the base class that this page's provisioner can support the
+     * HasAnyPermissionOrUserSiteMatch filter.
+     *
+     * @return ?array of site permissions or null
+     */
+    public function allSitePermissionNames() : ?array
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return array
+     */
+    protected function getFieldOptions() : array
+    {
+        $sites       = array_values(\Utility::getSiteList(false));
+        $visitlabels = array_values(\Utility::getVisitList());
+        $projects    = array_values(\Utility::getProjectList());
+        $subprojects = array_values(\Utility::getCohortList());
```

Comment:
Getting this error when loading the module
`[Wed Mar 15 14:58:43.129553 2023] [php:error] [pid 8415] [client ::1:38924] PHP Fatal error:  Uncaught Error: Call to undefined method Utility::getCohortList() in /var/www/loris/modules/schedule_module/php/schedule_module.class.inc:70\nStack trace:\n#0 /var/www/loris/php/libraries/DataFrameworkMenu.class.inc(154): LORIS\\schedule_module\\Schedule_Module->getFieldOptions()\n#1 /var/www/loris/php/libraries/NDB_Menu_Filter.class.inc(648): DataFrameworkMenu->toJSON()\n#2 /var/www/loris/php/libraries/NDB_Page.class.inc(753): NDB_Menu_Filter->display()\n#3 /var/www/loris/php/libraries/DataFrameworkMenu.class.inc(173): NDB_Page->handle()\n#4 /var/www/loris/php/libraries/NDB_Page.class.inc(730): DataFrameworkMenu->handle()\n#5 /var/www/loris/php/libraries/Module.class.inc(387): NDB_Page->process()\n#6 /var/www/loris/src/Middleware/ResponseGenerator.php(50): Module->handle()\n#7 /var/www/loris/src/Middleware/AuthMiddleware.php(63): LORIS\\Middleware\\ResponseGenerator->process()\n#8 /var/www/loris/src/Router/ModuleRouter.php(74): LORIS\\Middleware\\AuthMiddleware->process()\n#9 /var/www/loris/src/Middleware/ExceptionHandlingMiddleware.php(54): LORIS\\Router\\ModuleRouter->handle()\n#10 /var/www/loris/src/Router/BaseRouter.php(126): LORIS\\Middleware\\ExceptionHandlingMiddleware->process()\n#11 /var/www/loris/src/Middleware/ResponseGenerator.php(50): LORIS\\Router\\BaseRouter->handle()\n#12 /var/www/loris/src/Middleware/ContentLength.php(52): LORIS\\Middleware\\ResponseGenerator->process()\n#13 /var/www/loris/htdocs/index.php(55): LORIS\\Middleware\\ContentLength->process()\n#14 {main}\n  thrown in /var/www/loris/modules/schedule_module/php/schedule_module.class.inc on line 70, referer: http://localhost:8080/schedule_module/
`